builder/block_builder.go

```diff
@@ -9,12 +9,14 @@ import (
 	"github.com/iotaledger/iota.go/v3/pow"
 )
 
+const (
+	defaultProtocolVersion = 2
+)
+
 // NewBlockBuilder creates a new BlockBuilder.
-func NewBlockBuilder(protoVersion byte) *BlockBuilder {
+func NewBlockBuilder() *BlockBuilder {
```

Comment:
why not use `protoParas *iotago.ProtocolParameters` as an initializer parameter, then you have all required values for later steps and you don't need to provide it again for PoW, etc.